### ISSUE TITLE
[CDC][SPI_DEVICE] W_MASYNC from tlul xbar_main to memory_2p in spi_device

### ISSUE THREAD

**Issue by jeoongp** [lowRISC/opentitan#13834]
![Screen Shot 2022-07-25 at 2 10 20 PM](https://user-images.githubusercontent.com/103145460/180875354-ef95eb6a-8b1e-4bdf-b0b4-f18270291fff.png)

The dual port memory in spi_device receives signals for A (write) port from u_xbar_main on main_clk domain.
They are causing W_MASYNC errors because of conflict with signals from IO_DIV4_CLK.
Are they also for retention or clamp for low power?
![Screen Shot 2022-07-25 at 2 22 10 PM](https://user-images.githubusercontent.com/103145460/180876276-a70842a4-133d-40fb-a1eb-c79ac0eb81e3.png)

**Comment by tjaychen** [comment]
The dual memory in spi_device has a mode where it can be written by software.
When it is being written by software, the "data" is on the `main_clk` and the control signals are the ones that are synchronized. 

So I'm guessing that's where you are getting the M_ASYNC, since the software data is on `main`, and the local configuration is on `IO_DIV4`.  Is that roughly what you are seeing?

**Comment by jeoongp** [comment]
@tjaychen yes. But, all A port signals are also on main : a_req, a_write, a_addr, a_wdata, a_wmask
![Screen Shot 2022-07-25 at 3 00 40 PM](https://user-images.githubusercontent.com/103145460/180881192-ca720133-5569-4931-bb0b-5258b4e9180d.png)

**Comment by tjaychen** [comment]
hmm... do you have the path for a_req? a_req at least should be coming out of a synchronized pointer / logic in IO_DIV4. 
The others I can understand.

**Comment by jeoongp** [comment]
![Screen Shot 2022-07-25 at 2 22 10 PM](https://user-images.githubusercontent.com/103145460/181072455-9789adb5-a126-4bb6-9052-bb3a407c1a78.png)
Path A comes from tlul main_clk. 
Path A : tl_i.a_valid is propagated to reg2hw.upload_addrfifo.re => sys_addrfifo_rready_i => combined with Path B. 
The combined path is connected to the port A without synchronization stage inside the sram arbiter.

Input rvalid_i is propagated to output req_o as follows.
https://github.com/lowRISC/opentitan/blob/master/hw/ip/prim/rtl/prim_fifo_async_sram_adapter.sv#L287

**Comment by tjaychen** [comment]
@jeoongp i had a look at this, let me know if i'm in the right ballpark.
When I originally looked at this, I assumed that the orange line coming out of the async fifo was the request signal itself.

Looking at this more, I'm pretty certain it's probably the address (could you confirm?)
The usage of the address "should" be qualified with the actual `a_valid` received by `spi_device`. 

So even though all that logic remains "orange" (ie, main_clk), it should really be on the `IO_DIV4` clock domain. Can you have a look? I think this is the usual issue we saw where the tool does not understand the downstream qualification of `valid` and the `payload`.  But if would be good if you can have a look and let me know if this is also the behavior you see.

**Comment by jeoongp** [comment]
![Screen Shot 2022-07-26 at 4 43 23 PM](https://user-images.githubusercontent.com/103145460/181131048-a170bf7d-496a-4c0e-a895-0a9f23a687dc.png)
The previous schematic does not show the detailed signals. This zoomed schematic shows how a_req comes from path A captured by the tool. But, let me check manually if this tool report is correct.

**Comment by tjaychen** [comment]
o that's not good...why does it think `a_valid` is on the main_clock domain? maybe there's some other issue. Are you able to trace that one?

**Comment by jeoongp** [comment]
As discussed with @tjaychen, when dev_select is qualified with a_valid, a_valid is not generated from rclk (clk_div4).. Since it's 0 depth, it's just a bypass signal of input w_valid from wclk. w_valid is a combo logic with rdata_o from prim_fifo_async in xbar_main.
Its data out seems not qualified because data output is directly connected to the block output without qualification with r_valid. So, the tool does not recognize the read data output on read clock domain.

We can add a qualification in prim_fifo_sync.

**Comment by tjaychen** [comment]
sounds good @jeoongp !
